protected/humhub/modules/tour/models/TourConfig.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2015 HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\tour\models;
+
+use humhub\modules\space\models\Membership;
+use humhub\modules\space\models\Space;
+use humhub\modules\tour\Module;
+use Yii;
+
+/**
+ * Configurations for the introduction tour
+ * Loads and manage the configuration files
+ * Default files are in @tour/config
+ *
+ * @since 1.18
+ */
+class TourConfig
+{
+    public const PAGE_DASHBOARD = 'interface';
+    public const PAGE_SPACES = 'spaces';
+    public const PAGE_PROFILE = 'profile';
+    public const PAGE_ADMINISTRATION = 'administration';
+    /**
+     * A unique name of the page
+     */
+    public const KEY_PAGE = 'page';
+    /**
+     * Optional. Set to false for hiding the page
+     */
+    public const KEY_IS_VISIBLE = 'is_visible';
+    /**
+     * The title of the page displayed in the widget
+     */
+    public const KEY_TITLE = 'title';
+    /**
+     * The Controller class of the page where the tour must be started
+     */
+    public const KEY_CONTROLLER_CLASS = 'controller_class';
+    /**
+     * URL of the page
+     */
+    public const KEY_URL = 'url';
+    /**
+     * Name of the page for the next page to show after the tour.
+     * Can be `null`
+     */
+    public const KEY_NEXT_PAGE = 'next_page';
+    /**
+     * Driver.js configuration and steps
+     * Available values: https://driverjs.com/docs/
+     */
+    public const KEY_DRIVER = 'driver';
+
+    /**
+     * Return an array of valid and visible page configs
+     * The array keys are the Page names
+     */
+    public static function get(): array
+    {
+        /* @var Module $module */
+        $module = Yii::$app->getModule('tour');
+
+        $tourConfigs = [];
+        foreach ($module->tourConfigFiles as $file) {
+            $config = require Yii::getAlias($file);
+            if (
+                static::isValidConfig($config)
+                && (!isset($config[self::KEY_IS_VISIBLE]) || $config[self::KEY_IS_VISIBLE])
+            ) {
+                $tourConfigs[$config[self::KEY_PAGE]] = $config;
+            }
+        }
+
+        return $tourConfigs;
+    }
+
+    public static function isValidConfig($config): bool
+    {
+        $isValid =
+            is_array($config)
+            && !empty($config[self::KEY_PAGE])
+            && !empty($config[self::KEY_TITLE])
+            && !empty($config[self::KEY_CONTROLLER_CLASS])
```

Comment:
Maybe Controller Class could be optional?

Reply:
@luke- I don't understand how it could be optional.
It's used by the `TourConfig::getCurrent()` method to know if there is an existent config on the current page to start the Tour.
So it's not defined, the Tour will nether start, so the config file is useless.
Or do you see a use case where it could not be defined?

Reply:
@marc-farre If I were to implement a tour for the Mail Module, for example, and specify the start URL as `mail/mail/index`, why would I need to also specify a controller class again? Isn't that redundant?

The only important thing is that I am on the specified route.
Or am I missing something here?

Reply:
@luke- I see. Yes, I will implement that. I was using the previous Tour module behavior, which were testing the current page with the Controller class only, and not the starting URL.
So why renaming `KEY_CONTROLLER_CLASS` to `KEY_REQUIRED_CONTROLLER_CLASS` if it's optional?

Reply:
@marc-farre It seems a bit clearer to me that if `REQUIRED_CONTROLLER_CLASS` is set, the controller must have this class. I find `CONTROLLER_CLASS` alone somewhat vague. Perhaps also `TOUR_ON_CONTROLLER_CLASS` would be an option.